src/index.js

```diff
@@ -0,0 +1,196 @@
+import syntaxJSX from "@babel/plugin-syntax-jsx";
+
+/**
+ * @param {import("@babel/core")} babelApi
+ */
+export default function ({ template, types: t }) {
+  /** @type {import("@babel/core").Visitor} */
+  const visitor = {
+    JSXElement(path) {
+      path.replaceWith(buildTemplate(path.scope, transformElement(path)));
+    },
+    JSXFragment(path) {
+      path.replaceWith(buildTemplate(path.scope, transformFragment(path)));
+    },
+  };
+
+  /**
+   * @param {import("@babel/traverse").Scope} scope
+   * @param {import("@babel/types").Expression} scope
+   */
+  function buildTemplate(scope, content) {
+    const ref = scope.generateUidIdentifier("templateReference");
+    scope
+      .getProgramParent()
+      .push({ id: t.cloneNode(ref), init: t.objectExpression([]) });
+
+    return template.expression.ast`ESXToken.template(${ref}, ${content})`;
+  }
+
+  function transformElement(path) {
+    /** @type {import("@babel/types").JSXElement} */
+    const node = path.node;
+
+    const jsxElementName = node.openingElement.name;
+    if (t.isJSXNamespacedName(jsxElementName)) {
+      throw path.buildCodeFrameError(
```

Comment:
Do you want to support `<a:b />`?

Reply:
As tag, yes … as string tag element type with tag “a:b” … would that work?